Add an option to allow configuration of the ILIAS root URI

tbongers-cat opened the pull request:
In certain proxy configurations, the request URL used by SEB to create a hash of the exam keys might be changed from HTTPS to HTTP when it reaches the ILIAS server. In this case, the check for correct keys might fail. This PR adds a new option to ignore this change and check the keys against valid hashes for both URL variants.

@kergomard Please have a look. This is in response to the behaviour of the plugin in combination with our test server.

tbongers-cat committed:
Add an option to allow switching between HTTP and HTTPS for the request URL

kergomard commented:
Thank you very much @tbongers-cat , especially for making this configurable!
I think though, we should go for another solution, make the ILIAS-Root configurable and then change the `ilSEBAccessChecker::retrieveFullUri()` function. My reasoning:
* Our goal is to make the plugin work behind a reverse-proxy.
* I'm no SysAdmin, but as far as I understand this, the reverse-proxy might not only change the protocol, but the whole URI.
* The working of ILIAS behind reverse proxies is in my experience flaky at best, so we cannot really rely on the information provided by ILIAS (your installation showing http as the proto in perma-links is one more indication for me).

I would thus add a configuration-text-field, that contains the URI as ILIAS sees it, you can then change it manually to something else. If you made a mistake you should be able to just delete the content, save the form, and the default will be shown again. We would then use this value to create the uri in `ilSEBAccessChecker::retrieveFullUri()`. What do you think?

Thank you very much and best,
@kergomard

tbongers-cat commented:
@kergomard Thanks for the feedback. Your version does indeed cover this situation a bit better, so I'll have a look into it and come up with a new proposal.

tbongers-cat committed:
SEB: Alternative solution for use with reverse proxies

Reverts previous proposal and adds a configuration option for adapting protocol, host and port

tbongers-cat commented:
As discussed I have switched the config option over to a text field that makes it possible to change the ILIAS root URI, which is later used for checking SEB keys. 
@kergomard Please have a look :)

kergomard commented:
Hi @tbongers-cat !
Thank you very much for updating the PR and sorry to have left you waiting for so long!
This now looks good to me. I will merge.
Best,
@kergomard
